dvc/config_schema.py

```diff
@@ -229,7 +246,17 @@ class RelPath(str):
     "machine": {
         str: {
             "cloud": All(Lower, Choices("aws", "azure")),
+            "region": All(
+                Lower, Choices("us-west", "us-east", "eu-west", "eu-north")
+            ),
+            "image": str,
+            "name": str,
+            "spot": Bool,
+            "spot_price": Number(precision=5),
```

Comment:
I think this can actually just be 
```suggestion
            "spot_price": Coerce(float),
```

The 5-decimal place limit in the provider docs is just because amazon's spot prices will only be defined up with to 5-decimal places. But specifying a higher precision number as the comparison/limit won't actually harm anything.